include/ttmlir/Dialect/TTIR/Transforms/Passes.td

```diff
@@ -134,6 +134,14 @@ def TTIROptimizer: Pass<"ttir-optimizer", "::mlir::ModuleOp"> {
   ];
 }
 
+def TTIRGatherPatternMatch : Pass<"convert-ttir-to-ttir", "::mlir::ModuleOp"> {
```

Comment:
Is the intent of this pass to convert wide TTIR to narrow TTIR?

fyi @sdjordjevicTT

Reply:
Please sync with @jserbedzijaTT regarding this pass

Reply:
There is no gather support in metal, but there is embedding support. We decided to keep StableHLO to TTIR layer thin, and then add a pass to convert ttir::gatherOp to ttir::embeddingoP

Reply:
Hi @ddilbazTT, @jserbedzijaTT just merged the initial decomposition pass:
https://github.com/tenstorrent/tt-mlir/pull/969

You can reuse it for your scenario.